drivers/exec/driver_test.go

```diff
@@ -160,7 +161,71 @@ func TestExecDriver_StartWaitStop(t *testing.T) {
 		status, err := harness.InspectTask(task.ID)
 		require.NoError(err)
 		require.Equal(drivers.TaskStateExited, status.State)
-	case <-time.After(1 * time.Second):
+	case <-time.After(10 * time.Second):
+		require.Fail("timeout waiting for task to shutdown")
+	}
+
+	require.NoError(harness.DestroyTask(task.ID, true))
+}
+
+func TestExecDriver_StartWaitStopKill(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	ctestutils.ExecCompatible(t)
+
+	d := NewExecDriver(testlog.HCLogger(t))
+	harness := drivers.NewDriverHarness(t, d)
+	task := &drivers.TaskConfig{
+		ID:   uuid.Generate(),
+		Name: "test",
+	}
+
+	taskConfig := map[string]interface{}{
+		"command": "/bin/bash",
+		"args":    []string{"-c", "echo hi; sleep 600"},
+	}
+	encodeDriverHelper(require, task, taskConfig)
+
+	cleanup := harness.MkAllocDir(task, false)
+	defer cleanup()
+
+	handle, _, err := harness.StartTask(task)
+	require.NoError(err)
+
+	ch, err := harness.WaitTask(context.Background(), handle.Config.ID)
+	require.NoError(err)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		result := <-ch
+
+		// /bin/bash typically ignores INT, so it will be killed eventually
+		require.Equal(int(unix.SIGKILL), result.Signal)
```

Comment:
This check is flaky related to my note on `proc.Wait()` note below.  result should either have a SIGKILL signal (if we captured the exit code) or get -2 exit code (if we get `no child processes` proc.Wait() error).

I could use feedback on to deal with the uncertainty for testing purposes.

Reply:
Hm, so I'm a little confused as to the intent of this test and how it relates to Wait.

I think the intent is to assert that SIGINT does *not* cause the process to exit so it blocks for the 2s timeout until getting SIGKILLed? And how we handle waiting returns non-deterministic results on SIGKILL?

I *think* a sufficient solution would be that exiting took >2s and returned a non-zero exit code. If our implementation doesn't offer the same exit code each time, then the best we can do is make sure we don't erroneously let users assume a process exited fine (code=0) when it didn't.

You also can't use `require` (or anything that calls t.FailNow`) from a non-main goroutine. You'll either have to use `assert` or a chan to return errors back to the main goroutine.

Reply:
The ultimate test wants to ensure that all processes get killed even if process doesn't honor SIGTERM.

You are correct on the Destroy and use of require -- I copied the other test `TestExecDriver_StartWaitStop` verbatim with minor modification - I'll fix that one too.